spark/src/main/scala/geotrellis/spark/io/hadoop/HadoopLayerWriter.scala

```diff
@@ -46,11 +45,45 @@ class HadoopLayerWriter[K: SpatialComponent: Boundable: JsonFormat: ClassTag, V:
       case e: Exception => throw new LayerWriteError(id).initCause(e)
     }
   }
+
+  def update(id: LayerId, rdd: RDD[(K, V)]) = {
+    try {
+      if (!attributeStore.layerExists(id)) throw new LayerNotExistsError(id)
+
+      implicit val sc = rdd.sparkContext
+      implicit val mdFormat = cons.metaDataFormat
+      val layerPath = new Path(rootPath,  s"${id.name}/${id.zoom}")
+      val header =
+        HadoopLayerHeader(
+          keyClass = classTag[K].toString(),
+          valueClass = classTag[V].toString(),
+          path = layerPath
+        )
+
+      val (existingHeader, existingMetaData, existingKeyBounds, existingKeyIndex, existingSchema) =
+        attributeStore.readLayerAttributes[HadoopLayerHeader, cons.MetaDataType, KeyBounds[K], KeyIndex[K], Unit](id)
+
+      if (existingHeader != header) throw new HeaderMatchError(id, existingHeader, header)
+
+      val boundable = implicitly[Boundable[K]]
+      val keyBounds = boundable.getKeyBounds(rdd.asInstanceOf[RDD[(K, V)]])
+
+      if (!boundable.includes(keyBounds.minKey, existingKeyBounds) || !boundable.includes(keyBounds.maxKey, existingKeyBounds))
+        throw new OutOfKeyBoundsError(id)
+
+      rddWriter.write(rdd, layerPath, existingKeyIndex)
+    } catch {
+      case e: HeaderMatchError[_] => throw e.initCause(e)
+      case e: OutOfKeyBoundsError => throw e.initCause(e)
+      case e: LayerNotExistsError => throw e.initCause(e)
+      case e: Exception => throw new LayerUpdateError(id).initCause(e)
+    }
+  }
```

Comment:
The most obvious to make a `isUpdatable` function, but it is only a sugar (not sure it is necessary), and not reduces code; to make it more generic is harder, due to lots of different types.